components/admin/formSteps/TaskDetailsForm.tsx

```diff
@@ -0,0 +1,251 @@
+import {
+  TASK_OPTIONS,
+  TWITTER_OPTIONS,
+  getDefaultValues,
+} from "@constants/admin";
+import React, { FunctionComponent, useState } from "react";
+import styles from "@styles/admin.module.css";
+import Button from "@components/UI/button";
+import QuizStep from "../taskSteps/quizStep";
+import DiscordStep from "../taskSteps/discordStep";
+import TwitterFwStep from "../taskSteps/twitterFwStep";
+import CustomStep from "../taskSteps/customStep";
+import TwitterRwStep from "../taskSteps/twitterRwStep";
+import DomainStep from "../taskSteps/domainStep";
+import Typography from "@components/UI/typography/typography";
+import { TEXT_TYPE } from "@constants/typography";
+
+type TaskDetailsFormProps = {
+  steps: StepMap[];
+  setShowTwitterOption: React.Dispatch<React.SetStateAction<number>>;
+  setSteps: React.Dispatch<React.SetStateAction<StepMap[]>>;
+  showTwitterOption: number;
+  handleTasksInputChange: (
+    e: React.ChangeEvent<HTMLInputElement>,
+    index: number
+  ) => void;
+  buttonLoading: boolean;
+  onSubmit: () => void;
+  isButtonDisabled: boolean;
+  deleteTasks: (index: number) => void;
+};
+
+const TaskDetailsForm: FunctionComponent<TaskDetailsFormProps> = ({
+  steps,
+  setShowTwitterOption,
+  setSteps,
+  showTwitterOption,
+  handleTasksInputChange,
+  buttonLoading,
+  onSubmit,
+  isButtonDisabled,
+  deleteTasks,
+}) => {
+  const [currentTask, setCurrenTask] = useState(0);
+  const renderTask = () => {
```

Comment:
Could you make this component outside of àtaskDetailsForm.tsx` to make it easier ?

Reply:
actually this function uses a lot of props inside this component to work.

Migrating to a new component might mean moving all those props to this new component. Since we only use it here do you think we should still make a new component?